docs/CONNECTOR_PROTOCOL.md

```diff
@@ -125,12 +125,19 @@ In addition to the connector index `.elastic-connectors`, we have an additional
   connector_id: string; -> ID of the connector document in .elastic-connectors
   status: string; -> Same enum as last_sync_status above except the null value
   error: string; -> Optional error message
+  trigger_method: string; -> How the job is triggered. Possible values are on-demand, scheduled.
   index_name: string; -> The name of the content index
-  worker_hostname: string; -> The hostname of the worker to run the sync job
-  indexed_document_count: number; -> Number of documents indexed in the sync job
-  deleted_document_count: number; -> Number of documents deleted in the sync job
-  created_at: date; -> The date/time when the sync job is created
-  completed_at: date; -> The data/time when the sync job is completed
+  worker_hostname: string; -> The hostname of the worker to run the job
+  indexed_document_count: number; -> Number of documents indexed in the job
+  indexed_document_volume: number; -> The volume (in bytes) of documents indexed in the job
+  deleted_document_count: number; -> Number of documents deleted in the job
+  metadata: object; -> Connector-specific metadata
```

Comment:
Is it better to call it `metadata` or `stats`?

Reply:
Both are pretty general terms. Can we come up with something more specific and descriptive? What are some examples that will go in here?

Reply:
It's connector specific. 

e.g. A crawler metadata can be:
1. crawl type
2. max crawl depth
3. crawl duration
4. number of crawled urls
5. number of crawled pages
6. avg response time
7. number of 4xx error
8. number of 5xx error
9. list of domains/seed urls/sitemap urls.

Confluence metadata can be:
1. number of spaces indexed
2. number of pages indexed
3. number of blogs indexed
4. number of attachments indexed
5. current processing space
6. number of 4xx error
7. number of 5xx error
8. etc..

Reply:
Most of these look numeric. What about a `counters` object? And let's use as generic of key names for the counters as possible. Kibana will not care about the difference between "number of crawled pages" and "total number of confluence documents indexed", but will want to display both in the same place, I expect. So `num_documents` might be better.

For the non-numerics:
- "crawl type" could be a generic `job_type` field, and we'll probably eventually want that for other service types as well, as we'll want to display it in Kibana. Keeping the field name consistent for this will make our lives a lot easier.
- "list of domains/seed urls/sitemap urls" this sounds like configuration, which @vidok will be working to add as part of his work, so you probably don't need to worry about here

This kind of design work and iteration might be easier to do in the implementation document, in the future.

Reply:
I would love to see those properties actually mapped as properties as well, rather than mapping the whole thing as 'object'. That way we could actually execute efficient searches on these properties--things like, "give me all jobs with more than 10 4xx" errors. That's a lot harder if we don't map these.

Reply:
> So `num_documents` might be better.

> "crawl type" could be a generic `job_type` field

> "list of domains/seed urls/sitemap urls" this sounds like configuration

> I would love to see those properties actually mapped as properties as well

Hi @seanstory @sphilipse , the generic stats are defined at the root level of `.elastic-connectors-sync-jobs`, e.g. `indexed_document_count`. All those example fields above are connector-specific, which can be from Crawler, or Confluence, or Dropbox. We have no way to know what are the available fields in advance, and it's up to connector implementation.

>This kind of design work and iteration might be easier to do in the implementation document, in the future.

Yes, it was designed to be a new field `metadata` of type `object. I'm just not sure of the name

> Most of these look numeric. What about a `counters` object?

`counters` implies it should be all numbers? so does `stats`.

Reply:
> `counters` implies it should be all numbers? so does `stats`.

True. I feel that "stats" conflates a bit with system-level information - like memory and cpu usage, threadcount, idle threads, etc, I would expect to find under "stats". "Counters" on the other hand is where I have experience of ETL frameworks tracking how many of X have been seen, and also how much time has been spent on Y. But it's just a suggestion.

Reply:
> ll those example fields above are connector-specific, which can be from Crawler, or Confluence, or Dropbox. We have no way to know what are the available fields in advance, and it's up to connector implementation.

We have some idea right now, though. We know what metadata our current connectors will report. If we have some way of generalising over that kind of information that would be helpful.

Reply:
>We know what metadata our current connectors will report. If we have some way of generalising over that kind of information that would be helpful.

This is still in [discussion](https://elastic.slack.com/archives/C03GAPTQ2JF/p1667375423236469), but even there is some info we can generalize, we should make it one of the general metadata like `indexed_document_count`, which should reside at the root level. 

Here, the `metadata` is for connector-specific metadata, which can be anything, similar to `configuration` in `.elastic-connectors`. I'm just not sure about the name.

Reply:
Honestly given all that 'metadata' is probably the best term at this point.